### DISCUSSION TITLE
How to enable compression in es plugin of fluentbit?

### DISCUSSION THREAD

**Discussion by badu013** [fluent/fluent-bit#6779] [Q&A]
Hi,

we are running fluent-bit v2.0.8 right now as side car with our application running in k8s. with below configuration

fluent-bit.conf
```
[SERVICE]
    Flush        5
    Log_Level    info
    Daemon       Off
    Log_Level    info
    Parsers_File parsers.conf
    HTTP_Server  Off
    HTTP_Listen  0.0.0.0
    HTTP_Port    2020
@INCLUDE current.conf
@INCLUDE output.conf
```
current.conf
```
[INPUT]
    Name             tail
    Path             /mnt/log/current
    Multiline        On
    Parser_Firstline backend
    Parser_1         stacktrace
    Buffer_Max_Size   5MB
    Buffer_Chunk_Size 256k
[FILTER]
    Name            grep
    Match           *
    Exclude         LogLevel DEBUG
    Exclude         LogLevel TRACE
```

output.conf
```
[OUTPUT]
    Name             es
    Match            *
    Host             elasticsearch-endpoint
    Port             9200
    Logstash_Format On
    Logstash_Prefix *****-prod-backend
    Logstash_DateFormat %Y.%m.%d
    Type            _doc
[FILTER]
    Name   record_modifier
    Match  *
    Record hostname ${HOSTNAME}
    Record service ${srv_name}
    Record appversion ${project_version}
```
we want to enable compression in elasticsearch (es) plugin which was recently added from v1.9. Please help in enabling that feature in the plugin.

**Comment by patrick-stephens** [comment]
The docs seem to cover it: https://docs.fluentbit.io/manual/pipeline/outputs/elasticsearch#configuration-parameters

`compress gzip` seems to be what you want I think? 

I'm a little confused by the 1.9 statement, this was not added until the 2.0 releases.

https://github.com/fluent/fluent-bit/pull/5557

There is a PR to backport it though but it's not part of a 1.9 release yet.

https://github.com/fluent/fluent-bit/pull/6877
